tests/test_data.py

```diff
@@ -137,6 +137,13 @@ def test_encoding(postgresql, data, execute):
     assert "blocking éléphant" in blocking.query
 
 
+def test_InvalidTextRepresentation(postgresql, data, execute):
+    """Test for issue #275"""
+    postgresql.execute("select '123' ~ '\\d+', 'Hello world!\n', pg_sleep(3)")
```

Comment:
Can you explain how this query triggers the error?

If I'm running ` psql -c "select '123' ~ '\\d+', 'Hello world!\n', pg_sleep(3);"` on one side, and, aside:
```python
>>> import psycopg2
>>> conn = psycopg.connect()
>>> conn.autocommit = True
>>> cur.execute("select query from pg_stat_activity"); cur.fetchall()
<psycopg.Cursor [TUPLES_OK] [IDLE] (database=denis) at 0x7f2fa674ff60>
[('',), ('',), ("select '123' ~ '\\\\d+', 'Hello world!\\n', pg_sleep(3);",), ('select query from pg_stat_activity',), ('select query from pg_stat_activity',), ('',), ('',), ('',)]
```
I see no problem. Am I missing something?

Reply:
It's a test that should have been included when we introduced the cast to bytea.

I'd like to add it to make sure the solution we choose covers this (I wonder 
what will append if we end up using `psycopg2.extensions.register_type(psycopg2.extensions.BYTES)` ).

I'll remove it later if it's useless.